src/createRequestBody.ts

```diff
@@ -0,0 +1,42 @@
+import { isExtractableFile, extractFiles, ExtractableFile } from 'extract-files'
+import { Variables } from './types'
+
+/**
+ * Duck type if NodeJS stream
+ * https://github.com/sindresorhus/is-stream/blob/3750505b0727f6df54324784fe369365ef78841e/index.js#L3
+ */
+const isExtractableFileEnhanced = (value: any): value is ExtractableFile | { pipe: Function } =>
+  isExtractableFile(value) ||
+  (value !== null && typeof value === 'object' && typeof value.pipe === 'function')
+
+/**
+ * Returns Multipart Form if body contains files
+ * (https://github.com/jaydenseric/graphql-multipart-request-spec)
+ *
+ * Otherwise returns JSON
+ */
+export default function createRequestBody(query: string, variables?: Variables) {
+  const { clone, files } = extractFiles({ query, variables }, '', isExtractableFileEnhanced)
+
+  if (files.size > 0) {
+    const form = new FormData()
```

Comment:
I'm using graphql-request in node env and tried using this implementation on the server in environment where FormData is not available. We might need to use something like https://www.npmjs.com/package/form-data for compatibility with Node env.

Reply:
I think Form-Data should be polyfilled just like `fetch`, but I'll add readable error if form-data is not defined and improve documentation

Reply:
Yeah, that's fair. Would be great if documentation would explain how to set this up for node.js.